File: jcv-core/src/main/java/com/ekino/oss/jcv/core/comparator/TemplatedURLComparator.java
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator;

import java.net.MalformedURLException;
import java.net.URL;

import com.ekino.oss.jcv.core.JsonValueComparator;

import org.skyscreamer.jsonassert.ValueMatcherException;

public class TemplatedURLComparator implements JsonValueComparator<String> {

    @Override
    public boolean hasCorrectValue(String actual, String expected) {
        try {
            String urlWithoutTemplate = actual.replaceAll("\\{\\?.+}$", "");
            //noinspection ResultOfMethodCallIgnored
            new URL(urlWithoutTemplate);
            return true;
        } catch (MalformedURLException e) {
            throw new ValueMatcherException("Value is not a valid templated URL", e, expected, actual);
        }
    }
}

File: jcv-core/src/main/kotlin/com/ekino/oss/jcv/core/comparator/TemplatedURLComparator.kt
/*
 * Copyright (c) 2019 ekino (https://www.ekino.com/)
 */
package com.ekino.oss.jcv.core.comparator

import com.ekino.oss.jcv.core.JsonValueComparator
import org.skyscreamer.jsonassert.ValueMatcherException
import java.net.MalformedURLException
import java.net.URL

class TemplatedURLComparator : JsonValueComparator<String> {

  override fun hasCorrectValue(actual: String?, expected: String?): Boolean {
    try {
      val urlWithoutTemplate = actual?.replace("\\{\\?.+}$".toRegex(), "")
      URL(urlWithoutTemplate)
      return true
    } catch (e: MalformedURLException) {
      throw ValueMatcherException("Value is not a valid templated URL", e, expected, actual)
    }
  }
}
